include/picongpu/plugins/openPMD/WriteSpecies.hpp

```diff
@@ -0,0 +1,367 @@
+/* Copyright 2014-2019 Rene Widera, Felix Schmitt, Axel Huebl,
+ *                     Alexander Grund, Franz Poeschel
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/plugins/ISimulationPlugin.hpp"
+#include "picongpu/plugins/openPMD/openPMDWriter.def"
+#include "picongpu/plugins/openPMD/writer/ParticleAttribute.hpp"
+#include "picongpu/plugins/output/WriteSpeciesCommon.hpp"
+#include "picongpu/simulation_defines.hpp"
+
+#include <pmacc/assert.hpp>
+#include <pmacc/compileTime/conversion/MakeSeq.hpp>
+#include <pmacc/compileTime/conversion/RemoveFromSeq.hpp>
+#include <pmacc/dataManagement/DataConnector.hpp>
+#include <pmacc/mappings/kernel/AreaMapping.hpp>
+#include <pmacc/particles/ParticleDescription.hpp>
+#include <pmacc/particles/operations/ConcatListOfFrames.hpp>
+#include <pmacc/particles/particleFilter/FilterFactory.hpp>
+#include <pmacc/particles/particleFilter/PositionFilter.hpp>
+#if( PMACC_CUDA_ENABLED == 1 )
+#    include <pmacc/particles/memory/buffers/MallocMCBuffer.hpp>
+#endif
+
+#include <boost/mpl/at.hpp>
+#include <boost/mpl/begin_end.hpp>
+#include <boost/mpl/find.hpp>
+#include <boost/mpl/pair.hpp>
+#include <boost/mpl/size.hpp>
+#include <boost/mpl/vector.hpp>
+#include <boost/type_traits.hpp>
+#include <boost/type_traits/is_same.hpp>
+
+
+namespace picongpu
+{
+namespace openPMD
+{
+    using namespace pmacc;
+
+    /** Write copy particle to host memory and dump to openPMD file
+     *
+     * @tparam T_Species type of species
+     *
+     */
+    template< typename T_SpeciesFilter, typename T_Species = T_SpeciesFilter >
+    struct WriteSpecies
+    {
+    public:
+        typedef typename T_SpeciesFilter::Species ThisSpecies;
```

Comment:
please use `using` instead of `typedef` in all new places

Reply:
done